packages/api/portfolioDrafts/application/createApplicationStep.ts

```diff
@@ -101,111 +63,22 @@ export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayPr
-    .map(performDataValidationOnOperator)
-    .reduce((accumulator, validationErrors) => accumulator.concat(validationErrors), []);
-  const applicationErrors: ApplicationStepValidationErrors[] = applicationStep.applications
-    .map(performDataValidationOnApplication)
-    .reduce((accumulator, validationErrors) => accumulator.concat(validationErrors), []);
-
-  return operatorErrors.concat(applicationErrors);
-}
-
-/**
- * Performs data validation on the given Application object
- * @param application the object containing data to validate
- * @returns a collection of application validation errors
- */
-export function performDataValidationOnApplication(application: Application): Array<ApplicationStepValidationErrors> {
-  const errors = Array<ApplicationStepValidationErrors>();
-  if (application.name.length < 4 || application.name.length > 100) {
-    errors.push({
-      applicationName: application.name,
-      invalidParameterName: "name",
-      invalidParameterValue: application.name,
-      validationMessage: ValidationMessage.INVALID_APPLICATION_NAME,
-    });
-  }
-  const environmentErrors: ApplicationStepValidationErrors[] = application.environments
-    .map(performDataValidationOnEnvironment)
-    .reduce((accumulator, validationErrors) => accumulator.concat(validationErrors), []);
-  const operatorErrors: ApplicationStepValidationErrors[] = application.operators
-    .map(performDataValidationOnOperator)
-    .reduce((accumulator, validationErrors) => accumulator.concat(validationErrors), []);
-
-  return errors.concat(environmentErrors, operatorErrors);
-}
-
-/**
- * Performs data validation on the given Environment object
- * @param environment the object containing data to validate
- * @returns a collection of application validation errors
- */
-export function performDataValidationOnEnvironment(environment: Environment): Array<ApplicationStepValidationErrors> {
-  const errors = Array<ApplicationStepValidationErrors>();
-  if (environment.name.length < 4 || environment.name.length > 100) {
-    errors.push({
-      environmentName: environment.name,
-      invalidParameterName: "name",
-      invalidParameterValue: environment.name,
-      validationMessage: ValidationMessage.INVALID_ENVIRONMENT_NAME,
-    });
-  }
-  const operatorErrors: ApplicationStepValidationErrors[] = environment.operators
-    .map(performDataValidationOnOperator)
-    .reduce((accumulator, validationErrors) => accumulator.concat(validationErrors), []);
-  return errors.concat(operatorErrors);
-}
-
-/**
- * Performs data validation on the given Operator object
- * @param operator the object containing data to validate
- * @returns a collection of operator validation errors
- */
-export function performDataValidationOnOperator(operator: Operator): Array<ApplicationStepValidationErrors> {
-  const errors = Array<ApplicationStepValidationErrors>();
-  if (operator.display_name.length < 1 || operator.display_name.length > 100) {
```

Comment:
You might need to add this validation to the yaml spec (add exclusiveMinimum: 1, exclusiveMaximum:100).

Reply:
I added the regex pattern to the spec, like we discussed.